### ISSUE TITLE
Bot takes 3 minutes to connect and fails with timeout

### ISSUE THREAD

**Issue by JayshreeM** [Ullink/simple-slack-api#185]
With simple slack api, using directconnection as follows, slack takes almost 3 minutes to establish initial connection.
SlackSession session = SlackSessionFactory.createWebSocketSlackSession("slack-bot-auth-token");
Is there any performance improvement can be done where it doesnot read all users/channels info at the start.
We have larger team with 4000+ users as of now.

**Comment by kbakhit** [comment]
A similar library been also facing this issue: https://github.com/slack-ruby/slack-ruby-client/issues/134. Seems like either timeout needs to be increased to 180+ seconds or try implement rtm.connect instead of rtm.start, not sure..
Any updates on this issue?

**Comment by kbakhit** [comment]
I figured this out. 
The library uses rtm.start API which sends a websocket URL that expires within 30 seconds. 
In large teams, SlackJSONSessionStatusParser would be spending a lot of time parsing the users/channels response before returning to allow the websocket URL connection.

A fix would be connecting to the websocket URL ASAP before doing anything else, and also implement rtm.connect API instead @bcorne @JayshreeM

**Comment by bcorne** [comment]
Yes, definitely, rtm.connect should be used. I'm looking forward to do that change
